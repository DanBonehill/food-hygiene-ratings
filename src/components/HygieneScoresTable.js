import React from 'react';
import {connect} from 'react-redux';

const HygieneScoresTable = ({scores, totalScoresCount}) => (
    <div>
        <tr>
            <th>Rating</th>
            <th>Percentage</th>
        </tr>
        {
            scores.filter((score) => score.value !== 0).map((score) => (
                <tr>
                    <td>{score.title}</td>
                    <td>{Math.round(score.value / totalScoresCount * 100)}%</td>
                </tr>
            ))
        }
    </div>
);

const mapStateToProps = (state) => ({
    scores: state.hygieneScores.calculatedScores,
    totalScoresCount: state.hygieneScores.allScores.length
});

export default connect(mapStateToProps)(HygieneScoresTable)