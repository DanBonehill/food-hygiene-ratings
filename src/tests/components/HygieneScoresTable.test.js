import React from 'react';
import {shallow} from 'enzyme';
import HygieneScoresTable from '../../components/HygieneScoresTable';
import scores from '../fixtures/scores';

describe('HygieneScoresTable snapshot', () => {
    test('should render correctly', () => {
        const totalScoresCount = 25;
        const wrapper = shallow(<HygieneScoresTable scores={scores} totalScoresCount={totalScoresCount}/>);
        expect(wrapper).toMatchSnapshot()
    })
});