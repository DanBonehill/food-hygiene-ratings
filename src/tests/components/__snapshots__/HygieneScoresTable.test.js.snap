// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HygieneScoresTable snapshot should render correctly 1`] = `
<div>
  <tr>
    <th>
      Rating
    </th>
    <th>
      Percentage
    </th>
  </tr>
  <tr>
    <td>
      Pass
    </td>
    <td>
      60
      %
    </td>
  </tr>
  <tr>
    <td>
      Improvement Required
    </td>
    <td>
      40
      %
    </td>
  </tr>
</div>
`;
